Add documentation for Snowflake keypair auth support (#30944)
hashicorp/vault: website/content/docs/updates/important-changes.mdx

# Important changes

**Last updated**: 2025-06-12

Always review important or breaking changes and remediation recommendations
before upgrading Vault.

## New behavior

## Key Pair Authentication for Snowflake DB Secrets Engine

| Change       | Affected version | Affected deployments
| ------------ | ---------------- | --------------------
| New behavior | 1.20.0           | any

> As of version 1.20.0, Vault has added support for [authenticating to Snowflake
> databases using key pairs](/vault/api-docs/secret/databases#configure-connection).
> Snowflake is disabling password authentication for all users in November 2025,
> and password authentication will eventually be removed in subsequent Vault releases.

Reviewer: ```suggestion
As of version 1.20.0, Vault supports
[Snowflake database authentication using key pairs](/vault/api-docs/secret/databases#configure-connection)
as an alternative to password authentication, which Snowflake plans to disable
in November 2025. Vault support for password authentication with Snowflake is
now deprecated and will be removed in a future release.
```

It sounds like we're effectively deprecating the associated functionality. We should also update the deprecations page with the deprecation notice date (JUN 2025) and expected EOL (even if it's something general like CYQ2 2026).

Author: Good call, @schavis! We made sure to include a deprecation notice earlier in the release cycle, so I can confirm that the deprecation notice has been posted to our docs!
https://developer.hashicorp.com/vault/docs/updates/deprecation#snowflake-db-password-authentication

> Vault currently does not support rotate root for key pairs. Users can manually rotate the key pairs
> by updating both the root configuration in Vault with the new private key and the public
> key associated with the user within Snowflake. For more information on rotating key pairs, please refer
> to the official [Snowflake documentation](https://docs.snowflake.com/en/user-guide/key-pair-auth#configuring-key-pair-rotation).

Reviewer: ```suggestion
Vault does not support automatic rotation for root key pairs.

To manually rotate key pairs, update the root configuration in Vault with:

- the new private key
- the public key of the associated Snowflake user

For more information on rotating key pairs, refer to the official
[Snowflake documentation](https://docs.snowflake.com/en/user-guide/key-pair-auth#configuring-key-pair-rotation).
```
